wagtail_ab_testing/views.py

```diff
@@ -601,23 +606,45 @@ def filter(self, qs, value):
 
 
 class AbTestingReportFilterSet(WagtailFilterSet):
+    name = django_filters.CharFilter(
+        lookup_expr="icontains", label=gettext_lazy("Name")
+    )
     page = SearchPageTitleFilter()
     first_started_at = django_filters.DateFromToRangeFilter(
         label=gettext_lazy("Started at"), widget=DateRangePickerWidget
     )
 
     class Meta:
         model = AbTest
-        fields = ["status", "page", "first_started_at"]
+        fields = ["name", "status", "page", "first_started_at"]
 
 
 class AbTestingReportView(ReportView):
-    template_name = "wagtail_ab_testing/report.html"
     title = gettext_lazy("A/B testing")
-    header_icon = ""
+    index_results_url_name = "wagtail_ab_testing_admin:report_results"
+    index_url_name = "wagtail_ab_testing_admin:report"
+    results_template_name = "wagtail_ab_testing/report.html"
+    header_icon = "people-arrows"
 
     filterset_class = AbTestingReportFilterSet
 
+    @property
+    def template_name(self):
+        # TODO: compatibility: remove `template_name` getter once Wagtail 6.2
+        # is the minimum supported version
+        if WAGTAIL_VERSION < (6, 2):
+            return "wagtail_ab_testing/_compat/report.html"
+
+        # On Wagtail 6.2 and above, we can return the default template name
+        # because Wagtail 6.2 will use `results_template_name`
+        return ReportView.template_name
```

Comment:
**Suggestion**: this was a bit of a mind bender just by looking at the code, but https://docs.wagtail.org/en/stable/releases/6.2.html#adjust-the-templates explains what's going on. Maybe add the link in the comment.

**Thought**: library maintenance is hard 😓

Reply:
Ah yes good suggestion. I've added a link and reworded some bits.